server/src/main/java/org/elasticsearch/rest/action/search/SearchResponseMetrics.java

```diff
@@ -65,15 +75,45 @@ private SearchResponseMetrics(LongHistogram tookDurationTotalMillisHistogram, Lo
         this.responseCountTotalCounter = responseCountTotalCounter;
     }
 
-    public long recordTookTime(long tookTime) {
-        tookDurationTotalMillisHistogram.record(tookTime);
+    public long recordTookTime(long tookTime, SearchUsage searchUsage) {
+        tookDurationTotalMillisHistogram.record(tookTime, getAttributes(searchUsage));
         return tookTime;
     }
 
-    public void incrementResponseCount(ResponseCountTotalStatus responseCountTotalStatus) {
-        responseCountTotalCounter.incrementBy(
-            1L,
-            Map.of(RESPONSE_COUNT_TOTAL_STATUS_ATTRIBUTE_NAME, responseCountTotalStatus.getDisplayName())
-        );
+    public void incrementResponseCount(ResponseCountTotalStatus responseCountTotalStatus, SearchUsage searchUsage) {
+        responseCountTotalCounter.incrementBy(1L, getAttributes(searchUsage, responseCountTotalStatus));
```

Comment:
I wonder what's the impact of this, naively I'd expect that this will result in a combinatorial explosion of the number of measurements we need to represent. It was ok before because the status is an enum with few values.

Reply:
The attributes are added as fields (labels.*) added to the document that represents the measurement. I don't see a combinatorial explosion but different fields added to the measurement, that can be used to drill down / filter.

Reply:
I see thanks, I still don't understand how useful this can be though. How will we able to drill down if we don't know the count? If I use a term query once, will it be reported the same as using a prefix query thousand times?

Reply:
> If I use a term query once, will it be reported the same as using a prefix query thousand times?

It will. The main idea is to categorize queries as in "this query has this query construct" in order to:
- Be able to understand about changes in vector searches performance (the first thing we'll use this for)
- Understand if there are specific query types that are returning errors
- Understand the elements that are present in queries that perform poorly

Reply:
How so? We increment the count for each request but that doesn't mean that each increment is sent as a single document. I guess they're some merging happening in the background and that the counter is updated only periodically.

Reply:
See https://opentelemetry.io/docs/specs/otel/metrics/supplementary-guidelines/
My understanding is that the attributes are merged so that measurements are sent per **unique** values of attributes. That's what I meant by combinatorial explosion, the number of measurements we'll send will grow significantly.

Reply:
I see - memory usage will increase as well as we need to remember every combination to send just the deltas.

I understand your concerns now. I'll close this PR because of its impact.

Thanks for your review @jimczi !